jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -127,6 +192,33 @@ default boolean isDirect()
         return getByteBuffer().isDirect();
     }
 
+    /**
+     * @return whether the {@code ByteBuffer} has remaining bytes left for reading
+     */
+    default boolean isEmpty()
+    {
+        return !hasRemaining();
+    }
+
+    /**
+     * @return whether the {@code ByteBuffer} has remaining bytes left for appending
+     */
+    default boolean isFull()
+    {
+        return space() == 0;
+    }
+
+    /**
+     * Copies the contents of this retainable byte buffer at the end of the given byte buffer.
+     * @param toInfillMode the destination buffer.
+     * @throws BufferOverflowException – If there is insufficient space in this buffer for the remaining bytes in the source buffer
+     * @see ByteBuffer#put(ByteBuffer)
+     */
+    default void putTo(ByteBuffer toInfillMode) throws BufferOverflowException
```

Comment:
There is already an `appendTo(RBB)`.    We only need a `putTo` because BB's can be modal and if you have one in fill mode then it doesn't need to be flipped.   RBBs are not modal, so there is no difference between an `appendTo(RBB)` and `putTo(RBB)`.... hmmmm maybe the put overflows, whilst the append limits with a boolean return... not sure that is worth an extra method?